src/core/DataFile.cpp

```diff
@@ -1355,6 +1355,35 @@ void DataFile::upgrade_1_3_0()
 	}
 }
 
+void DataFile::upgrade_noHiddenClipNames()
+{
+	QDomNodeList tracks = elementsByTagName("track");
+
+	auto clearDefaultNames = [](QDomNodeList clips, QString trackName)
+	{
+		for (int j = 0; !clips.item(j).isNull(); ++j)
```

Comment:
Wouldn't `j < clips.length()` be a more natural choice of condition?

Reply:
I believe this iteration code is nicked directly from another upgrade routine, but I can swap it for a comparison to `clips.size()`. I'm tempted to add `if (item(j).isNull) { continue(); }` if so (to avoid introducing additional assumptions), but maybe that's paranoid?

Reply:
How about using `clips.size()` and checking with `clips.item(j).isElement()` before converting to `QDomElement`?

Reply:
Ok, after further investigation it looks like the check is unnecessary. I added the following at the start of the upgrade routine:

```
qDebug() << "About to test null stuff, if we crash after this you know why";
QDomNode nullNode = QDomNode();
QDomElement nullElement = nullNode.toElement();
QString nullString = nullElement.attribute("name", "null");
qDebug() << "nullString = " << nullString;
```

Upon opening an older project, console output is: 

```
About to test null stuff, if we crash after this you know why
nullString =  "null"
```

So if the node is null we'll receive a null element, and calling `attribute` on the null element will fall back to the default string. In this case `clipName` will be set to an empty string, which is harmless.